pkg/cmd/cli/cmd/generate.go

```diff
@@ -62,21 +63,21 @@ func NewCmdGenerate(cfg *api.Config) *cobra.Command {
 		Short: "Generate a Dockerfile based on the provided builder image",
 		Example: `
 # Generate a Dockerfile from a builder image:
-$ s2i generate docker://docker.io/centos/nodejs-10-centos7 Dockerfile.gen
+$ s2i generate docker.io/centos/nodejs-10-centos7 Dockerfile.gen
 `,
 		RunE: func(cmd *cobra.Command, _ []string) error {
 			if cmd.Flags().NArg() != 2 {
 				return cmd.Help()
 			}
 
-			ref, err := alltransports.ParseImageName(cmd.Flags().Arg(0))
+			cfg.BuilderImage = cmd.Flags().Arg(0)
+			cfg.AsDockerfile = cmd.Flags().Arg(1)
+
+			ref, err := alltransports.ParseImageName("docker://" + cfg.BuilderImage)
```

Comment:
What does `ParseImageName` do if there is no protocol? Ideally we should be able to support both forms.

`BuilderImage` needs to be a valid pull spec that is used in a Dockerfile.

Reply:
> What does ParseImageName do if there is no protocol?

It gets us the `ref` for doing the following later on
```
img, err := ref.NewImage(ctx, &types.SystemContext{})
	if err != nil {
		return nil, err
	}

	imageMetadata, err := img.Inspect(ctx)
	if err != nil {
		return nil, err
	}
```

> Ideally we should be able to support both forms.

We don't support the following at the moment, but we could support `docker://..` in `s2i generate.. ` of course ?
```
s2i build https://github.com/openshift/ruby-hello-world docker://docker.io/centos/ruby-22-centos7
```

Reply:
I think we should assume `docker://` as default, it's the default approach in `s2i` in general. Therefore, we need a parser to split up container image address and protocol, on which it would assume `docker://` when protocol is empty.